### PR TITLE
User friendly unsupported feature notification

### PR DESCRIPTION
**Pull request by Yuqi-Du**
**What this PR does**: 
JSONAPI relies on Bridge to get features that Cassandra backends support.
Add friendly unsupported feature notifications.
Shut down JSONAPI when SAI is not enabled.

**Which issue(s) this PR fixes**:
Fixes #<389>

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by Yuqi-Du** [comment]
Have error when building. `Error:  The forked VM terminated without properly saying goodbye. VM crash or System.exit called?`
May change System.exit(0).

**Comment by Yuqi-Du** [comment]
Have changed system.exit, now use Quarkus.asyncExit()

**Comment by Yuqi-Du** [comment]
Currently, Tests all fail, that is because in the main repo, DSE-NEXT persistence does not enable SAI, so JSONAPI will shutdown.
Have a [PR](https://github.com/stargate/stargate/pull/2732) for enabling SAI for DSE-NEXT persistence.

**Review by maheshrajamani — APPROVED** [review]
LGTM.

File: src/main/java/io/stargate/sgv2/jsonapi/JsonApiStartUp.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  private static final Logger LOGGER = LoggerFactory.getLogger(JsonApiStartUp.class);
+  private final DataStoreProperties dataStoreProperties;
+
+  @Inject
+  public JsonApiStartUp(DataStoreProperties dataStoreProperties) {
+    this.dataStoreProperties = dataStoreProperties;
+  }
+
+  void onStart(@Observes StartupEvent ev) {
+    LOGGER.info(
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/JsonApiStartUp.java:24]: SLF4J loggers have their own formatting so use of `String.format()` is not necessary. You can just use:

    LOGGER.info("VectorSearch Enabled: {}", dataStoreProperties.vectorSearchEnabled());

which avoids evaluate if logger not enabled, so bit more efficient. But more importantly bit easier to read, less to type.

**Pull request merged**